### PR TITLE
Update parallel.rst

### PR DESCRIPTION
**Pull request by simeonbassett**
It was very confusing to me how to pass messages back and forth between processes, so I thought a complete example would help quite a bit.

### CODE REVIEW

**Comment by vtjnash** [comment]
looks like this failed the whitespace test on CI

File: doc/stdlib/parallel.rst [hunk 1]
```diff
@@ -234,6 +234,18 @@ General Parallel Computing Support
...
 
+       procList = addprocs(1)              #  add another process
+       rr  =  RemoteRef(procList[1])       #  create a remote reference on the next process
+       @everywhere function f(inRR::RemoteRef)
+         x  =  take!(inRR)
+         put!(inRR, x ^ 2)
+       end
+       @spawnat procList[1] f(rr)
+       put!(rr, 8)
+       @show( take!(rr) )  #   =>  64
```
Comment (amitmurthy) [inline doc/stdlib/parallel.rst]: Though the chances of seeing a return value of 8 here is almost nil, julia does not guarantee that `f` will execute on the remote worker before the `take!` here.

Hence, a better example for message passing would be to have two remote refs, say r1 and r2. r1 is always written to from the local process and read from the remote process, while r2 is always written to at the remote process and read from the local.

**Comment by jakebolewski** [comment]
@simeonbassett do you have time to squash and rebase?  Also, it would be great if you could address @amitmurthy's comment.

**Comment by quinnj** [comment]
Old Doc PRs tend to be the most stale.....feel free to open additional issues/PRs around improvements on latest master.

**Pull request closed without merging**